src/ec-evaluator/index.ts

```diff
@@ -23,9 +24,11 @@ export const runECEvaluator = (
       resolve({ status: 'finished', context, value } as Finished);
     });
   } catch (e) {
-    context.errors.push(e);
+    // Possible interpreting language error thrown, so conversion to RuntimeError may be required.
+    const error = e.type ? e : new RuntimeError(e.message);
+    context.errors.push(error);
     return new Promise((resolve, _) => {
-      resolve({ status: 'error' } as Error);
+      resolve({ status: 'error', context } as Error);
```

Comment:
Do you really need to type cast the stuff in `resolve` as an Error? 
Also, it might be better to just use `async` for this whole function rather than the older `Promise()` constructor.

Reply:
Actl neither `parse()` and `evaluate()` is declared `async` (not sure if they should), and the return of `Promise<Result>` by the entry function is adopted from `js-slang`. However, `runECEvaluator()` most probably won't be the entry function anyway (might need to declare another function that integrates all our projs from `frontend`), so let's leave this for now?